lib/sqlalchemy/sql/elements.py

```diff
@@ -4307,11 +4429,11 @@ def _tq_label(self) -> Optional[str]:
         return self._gen_tq_label(self.name)
 
     @HasMemoized.memoized_attribute
-    def _render_label_in_columns_clause(self):
+    def _render_label_in_columns_clause(self) -> bool:
         return True
 
     @HasMemoized.memoized_attribute
-    def _non_anon_label(self):
+    def _non_anon_label(self) -> Optional[str]:
```

Comment:
Same as `_render_label_in_columns_clause()` here. Thinking about these decorators now - seems like this problem only occurs in decorated functions

Reply:
```py
@propertydef _non_anon_label(self) -> Optional[str]:
```

Reply:
So the error is: `Signature of "_non_anon_label" incompatible with supertype "ColumnElement"  [override]  mypy`.

If you look at the signature of the method on `ColumnElement` and replace `@HasMemoized.memoized_attribute` on child method with `@property` the error is gone.
https://github.com/GodelTech/sqlalchemy/blob/ced96d1f1d75a5a0171bf5942d497011f8fdb297/lib/sqlalchemy/sql/elements.py#L1321-L1322
Any other combination leaves the error in place, so looks like there's currently no way to type annotate such cases without rethinking how methods are defined. So just `# type: ignore` this and other cases and let the author now the problem in the future PR.